### PR TITLE
[HW] Add config file for Character class

### PR DESCRIPTION
**Pull request by juanbono**

### CODE REVIEW

File: example.yaml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+      base: 15
+      increase: 3
+    strength:
+      base: 8
+      increase: 2
+    speed:
+      base: 12
+      increase: 2
+    distance:
+      Near: 9
```
Comment (facundoolano) [inline example.yaml:36]: I think we shouldn't have to specify the distance like this (for one the number is not used). We should consider:

1. have a standalone field like `type: near` that gets translated to the group the enemies end up in
2. have sublists to group the enemies, e.g.
```yaml
enemies:
  -  near:
    - wolf: ...
    - rat: ...
  - middle:
    - ...
```
3. (if it's easier to implement) just skip the enemy grouping for now and have it hardcoded in the code.

File: example.yaml [hunk 2]
```diff
@@ -0,0 +1,216 @@
+hero:
```
Comment (facundoolano) [inline example.yaml:1]: Consider this shorter format:

```yaml
player:
  - hero:
      hp: [30, 7]
      str: [12, 3]
      spd: [11, 2]
enemy:
  - rat:
      type: common
      hp: [10, 3]
      str: [5, 2] 
      spd: [16, 2]
  - wolf:
      type: common
      hp: [15, 3]
      strength: [8, 2]
      speed: [12, 2]
  - skeleton:
      type: rare
      hp: [30, 3]
      str: [10, 2]
      spd: [10, 2]
  - balrog:
      type: legendary
      hp: [200, 3]
      str: [200, 2]
      spd: [14, 2]
```

In addition to being shorter, you can keep the shorter Stat struct

File: src/character/config.rs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+#[derive(Debug, Serialize, Clone, Deserialize)]
+pub struct CharacterConfig {
+    pub hero: Class,
+    pub enemies: Vec<Class>,
+}
+
+impl CharacterConfig {
+    pub fn load(file: path::PathBuf) -> Result<Self, Box<dyn std::error::Error>> {
+        let f = fs::File::open(file)?;
```
Comment (facundoolano) [inline src/character/config.rs:15]: One thing that's kind of important for this feature is that we shouldn't force the user to setup their own config file (we let them do it if they want, but we need to fallback to the default definition).

This means we need to explore how to package the yaml with default classes into the binary. The approach I was thinking of is using [include_bytes](https://doc.rust-lang.org/std/macro.include_bytes.html) or some crate with a similar effect. Then we look for the custom classes file at a known location, e.g. `~/.rpg/config.yml` but if that is missing we use the defaults.

File: src/game/mod.rs [hunk 4]
```diff
@@ -44,6 +45,7 @@ impl Game {
     pub fn load() -> Result<Self, Error> {
         let data = fs::read(data_file()).or(Err(Error::NoDataFile))?;
         let game: Game = bincode::deserialize(&data).unwrap();
+        let _config = character::CharacterConfig::load(config_file());
```
Comment (facundoolano) [inline src/game/mod.rs:48]: consider doing this directly in main, so tha game module doesn't need to know about it

File: example.yaml (hunk 2 above)
Comment (juanbono) [reply example.yaml:1]: yes, i'm working on this :)

Comment (juanbono) [reply example.yaml:1]: done!

**Pull request closed without merging**